.github/workflows/test.yml

```diff
@@ -11,18 +11,24 @@ permissions:
   contents: read
 
 jobs:
-  integration:
-    name: Integration tests
+  build:
+    name: Build
     strategy:
       fail-fast: false
       matrix:
-        # macos-13-large is used as macos-13 seems too flaky.
-        # macos-14 (ARM) and later cannot be used for the most part of the job
-        # due to the lack of the support for nested virt.
-        #
-        # TODO: add macos-15-large https://github.com/lima-vm/socket_vmnet/pull/63
-        platform: [macos-13-large, macos-14-large]
+        platform: [macos-13, macos-14, macos-15]
     runs-on: ${{ matrix.platform }}
+    timeout-minutes: 10
+    steps:
+      - uses: actions/checkout@eef61447b9ff4aafe5dcd4e0bbf5d482be7e7871 # v4.2.1
+        with:
+          fetch-depth: 1
+      - name: Build
+        run: make
+  integration:
+    name: Integration tests
+    # Only a single version of macOS is used for running integration tests, due to the flakiness of the CI.
+    runs-on: macos-15-large  # Intel (supports nested virtualization)
     timeout-minutes: 40
     steps:
       - uses: actions/checkout@eef61447b9ff4aafe5dcd4e0bbf5d482be7e7871 # v4.2.1
```

Comment:
This compiles both for x86_64 and aarch64, but we test only on x86_64 - any reason to build twice when we test the build in the build job?

Reply:
OK, makes sense, this is also very quick.